submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt

```diff
@@ -45,6 +58,159 @@ internal class SubmissionMongoQueryServiceTest(
+        }
+
+        @Test
+        fun `filtered by to release time`() {
+            val matchDate = OffsetDateTime.of(2000, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+            val mismatchDate = OffsetDateTime.of(2010, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+
+            saveAsRequest(fullExtSubmission.copy(accNo = "accNo1", releaseTime = matchDate, section = section))
+            saveAsRequest(fullExtSubmission.copy(accNo = "accNo2", releaseTime = mismatchDate, section = section))
+            submissionRepo.save(
+                testDocSubmission.copy(accNo = "accNo3", releaseTime = matchDate.toInstant(), status = PROCESSED)
+            )
+
+            val result = testInstance.getSubmissionsByUser(
+                SUBMISSION_OWNER,
+                SubmissionFilter(rTimeTo = OffsetDateTime.of(2005, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC), limit = 3)
+            )
+
+            assertThat(result).hasSize(2)
+            assertThat(result.first().accNo).isEqualTo("accNo1")
+            assertThat(result.second().accNo).isEqualTo("accNo3")
+        }
+
+        @Test
+        fun `filtered by released`() {
+            saveAsRequest(fullExtSubmission.copy(accNo = "accNo1", released = true, section = section))
+            saveAsRequest(fullExtSubmission.copy(accNo = "accNo2", released = false, section = section))
+            submissionRepo.save(testDocSubmission.copy(accNo = "accNo3", released = true, status = PROCESSED))
+
+            val result = testInstance.getSubmissionsByUser(
+                SUBMISSION_OWNER,
+                SubmissionFilter(released = true, limit = 2)
+            )
+
+            assertThat(result).hasSize(2)
+            assertThat(result.first().accNo).isEqualTo("accNo1")
+            assertThat(result.second().accNo).isEqualTo("accNo3")
+        }
+
+        @Test
+        fun `get greatest version submission`() {
+            val sub1 = submissionRepo.save(testDocSubmission.copy(accNo = "accNo1", version = 3, status = PROCESSED))
+            submissionRepo.save(testDocSubmission.copy(accNo = "accNo1", version = -2, status = PROCESSED))
+            submissionRepo.save(testDocSubmission.copy(accNo = "accNo1", version = -1, status = PROCESSED))
+
+            val result = testInstance.getSubmissionsByUser(
+                SUBMISSION_OWNER,
+                SubmissionFilter(accNo = "accNo1", limit = 3)
+            )
+
+            assertThat(result).hasSize(1)
+            assertThat(result.first()).isEqualTo(sub1.asBasicSubmission())
+        }
+
+        private fun saveAsRequest(extSubmission: ExtSubmission): ExtSubmission {
+            requestRepository.saveRequest(asRequest(extSubmission))
+            return extSubmission
+        }
+
+        private fun asRequest(submission: ExtSubmission) = SubmissionRequest(
+            accNo = submission.accNo,
+            version = submission.version,
+            submission = BasicDBObject.parse(ExtSerializationService().serialize(submission))
```

Comment:
There's no neeed to create a new serialization service instance

Reply:
great catch